x-pack/plugins/search_indices/public/plugin.ts

```diff
@@ -81,8 +81,12 @@ export class SearchIndicesPlugin
     docLinks.setDocLinks(core.docLinks.links);
     if (this.pluginEnabled) {
       indexManagement?.extensionsService.setIndexDetailsPageRoute({
-        renderRoute: (indexName) => {
-          return `/app/elasticsearch/indices/index_details/${indexName}`;
+        renderRoute: (indexName, detailsTabId) => {
+          const route = `/app/elasticsearch/indices/index_details/${indexName}`;
+          if (detailsTabId && SearchIndexDetailsTabValues.includes(detailsTabId)) {
```

Comment:
Can renderRoute detailsTabId param be a `SearchIndexDetailsTabs` type instead of a string?

Reply:
updated in https://github.com/elastic/kibana/pull/196787/commits/9fc3957d20ecc575998da3859a905543c7e5848b

Reply:
After some thoughts,  I think current implementation -` detailsTabId && SearchIndexDetailsTabValues.includes(detailsTabId)` is good. 

We are searching detailsTabId value in SearchIndexDetailsTabs keys whereas the [search_indices routes](https://github.com/elastic/kibana/blob/main/x-pack/plugins/search_indices/public/components/indices/indices_router.tsx#L22-L30) are mapped based on the values. When a new tab id added in search_indices we need to ensure that tabId keys and values are same and are in Uppercase to work this logic. Which i think is a bit brittle. WDYT?

Reply:
Yes, that sounds like the right direction for both.

Reply:
Reverting https://github.com/elastic/kibana/commit/9fc3957d20ecc575998da3859a905543c7e5848b